### PR TITLE
ADPQVP2018-136 As an admin, I should be able to Deactivate/Activ…

### PR DESCRIPTION
**Pull request by RakshitRajeevintimetec**
…ate users in the user settings.

Code Coverage for User Controller is 80%.

### CODE REVIEW

File: server-src/src/test/java/com/itt/test/kmt/services/UserServiceTests.java [hunk 1]
```diff
@@ -170,6 +170,46 @@ public final void updateUser() {
...
+
+        // Act
+        User user = userService.changeUserStatus(user1.getId(), false);
+
+        // Assert
+        assertEquals(user.isActive(), user1.isActive());
+
+        verify(userRepository, times(1)).save(user1);
+    }
+
```
Comment (koteswara-v) [inline server-src/src/test/java/com/itt/test/kmt/services/UserServiceTests.java]: Can you add one more test case where user's status is true and calling this service with 'true' again

File: server-src/src/test/resources/UserTestData.json [hunk 2]
```diff
@@ -24,6 +24,6 @@
             "firstName": "Rakshit",
             "userRole":"user",
             "password":"password@123",
-            "active"  : true
+            "active"  : false
```
Comment (koteswara-v) [inline server-src/src/test/resources/UserTestData.json]: Create user with active status - true, by default.

File: server-src/src/main/java/com/itt/kmt/services/UserService.java [hunk 3]
```diff
@@ -73,6 +73,24 @@ public void deleteUserById(final String id) {
...
+    public User changeUserStatus(final String id, final boolean isActive) {
+       User existingUser = repository.findOne(id);
+
+       if (existingUser != null && existingUser.isActive() != isActive) {
+           existingUser.setActive(isActive);
+           return repository.save(existingUser);
+        } else if (existingUser == null) {
+            throw new RuntimeException("user with the id does not exist");
+        } else {
+            throw new RuntimeException("Operation not supported");
```
[the remark this answers is not in the document]
Comment (RakshitRajeevintimetec) [reply server-src/src/main/java/com/itt/kmt/services/UserService.java]: done.

File: server-src/src/main/java/com/itt/kmt/services/UserService.java [hunk 4]
```diff
@@ -73,6 +73,24 @@ public void deleteUserById(final String id) {
...
+        }
+    }
     /**
      * Activates inactive User.
```
[the remark this answers is not in the document]
Comment (RakshitRajeevintimetec) [reply server-src/src/main/java/com/itt/kmt/services/UserService.java]: done.

File: server-src/src/test/resources/UserTestData.json (hunk 2 above)
Comment (RakshitRajeevintimetec) [reply server-src/src/test/resources/UserTestData.json]: done.

File: server-src/src/test/java/com/itt/test/kmt/services/UserServiceTests.java (hunk 1 above)
Comment (RakshitRajeevintimetec) [reply server-src/src/test/java/com/itt/test/kmt/services/UserServiceTests.java]: done.

**Pull request merged**